presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -493,7 +496,12 @@ private void enforceBroadcastMemoryLimit(long allocated, long delta, long maxMem
     private void enforceUserMemoryLimit(long allocated, long delta, long maxMemory)
     {
         if (allocated + delta > maxMemory) {
-            throw exceededLocalUserMemoryLimit(succinctBytes(maxMemory), getAdditionalFailureInfo(allocated, delta), heapDumpOnExceededMemoryLimitEnabled, heapDumpFilePath);
+            if (hasLowPartitionCount()) {
```

Comment:
can you gate this with a configuration property?  Changing the error type for such a common error can cause tools that try to handle it to start failing?

Reply:
We are not changing error type. We just use different error code.